tensorflow/python/ops/random_ops.py

```diff
@@ -310,21 +310,29 @@ def random_crop(value, size, seed=None, name=None):
   # If size > value.shape, zero pad the result so that it always has shape
   # exactly size.
   with ops.name_scope(name, "random_crop", [value, size]) as name:
+    size_dim = size
     value = ops.convert_to_tensor(value, name="value")
     size = ops.convert_to_tensor(size, dtype=dtypes.int32, name="size")
     shape = array_ops.shape(value)
-    check = control_flow_ops.Assert(
-        math_ops.reduce_all(shape >= size),
-        ["Need value.shape >= size, got ", shape, size],
-        summarize=1000)
-    shape = control_flow_ops.with_dependencies([check], shape)
-    limit = shape - size + 1
-    offset = random_uniform(
-        array_ops.shape(shape),
-        dtype=size.dtype,
-        maxval=size.dtype.max,
-        seed=seed) % limit
-    return array_ops.slice(value, offset, size, name=name)
+    if value.shape[0].value < size_dim[0]:
```

Comment:
value.shape might be None if value has a fully unknown shape; this needs to handle the static and dynamic shapes

Reply:
also .value in shapes will not work in tf with v2 behavior enabled; use compat.dimension_value() instead

Reply:
Thanks @alextp. I've updated using compat.dimension_value(). But, I'm not sure how to handle when value.shape is None. Do you have any suggestions?